Q/Qt/build_tarballs.jl

```diff
@@ -0,0 +1,154 @@
+            -prefix $prefix $commonoptions \
+            -skip qtwinextras
+        ;;
+        
+	*w64-mingw*)
+        apk add g++ linux-headers
+        export PATH=$(echo "$PATH" | sed -e 's/\/opt\/bin://')
+        
+        ../qt-everywhere-src-5.14.1/configure -platform linux-g++ -xplatform win32-g++ -device-option CROSS_COMPILE=/opt/bin/$target- \
+            -prefix $prefix $commonoptions \
+            -opengl desktop
+		;;
+
+    *arm-linux*)
+        apk add g++ linux-headers
+        export PATH=$(echo "$PATH" | sed -e 's/\/opt\/bin://')
+        export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$prefix/share/pkgconfig
+        export PKG_CONFIG_LIBDIR=$prefix/lib/pkgconfig
+        
+        ../qt-everywhere-src-5.14.1/configure QMAKE_LFLAGS=-liconv -platform linux-g++ -device linux-rasp-pi3-g++ -device-option CROSS_COMPILE=/opt/bin/$target- \
+            -extprefix $prefix $commonoptions \
+            -skip qtwinextras -system-xcb -fontconfig -sysroot /opt/$target/bin/../$target/sys-root
+		;;
+esac
+
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Linux(:armv7l, libc=:glibc, call_abi=:eabihf),
+    Windows(:x86_64),
+    Linux(:x86_64, libc=:glibc)
+]
+platforms = expand_cxxstring_abis(platforms)
+
+platforms_macos = [ MacOS(:x86_64) ]
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct(["Qt5Core", "libQt5Core", "QtCore"], :libqt5core),
+    LibraryProduct(["Qt5Quick", "libQt5Quick", "QtQuick"], :libqt5quick),
+    LibraryProduct(["Qt5QuickControls2", "libQt5QuickControls2", "QtQuickControls2"], :libqt5quickcontrols2),
+    LibraryProduct(["Qt5Svg", "libQt5Svg", "QtSvg"], :libqt5svg),
+    LibraryProduct(["Qt5Widgets", "libQt5Widgets", "QtWidgets"], :libqt5widgets),
+]
+
+products_macos = [
+    FrameworkProduct("QtCore", :libqt5core),
+    FrameworkProduct("QtQuick", :libqt5quick),
+    FrameworkProduct("QtQuickControls2", :libqt5quickcontrols2),
+    FrameworkProduct("QtSvg", :libqt5svg),
+    FrameworkProduct("QtWidgets", :libqt5widgets),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("Xorg_libX11_jll"),
+    Dependency("Xorg_libXext_jll"),
+    Dependency("Xorg_glproto_jll"),
+    Dependency("Xorg_libxcb_jll"),
+    Dependency("Xorg_xcb_util_wm_jll"),
+    Dependency("Xorg_xcb_util_image_jll"),
+    Dependency("Xorg_xcb_util_keysyms_jll"),
+    Dependency("Xorg_xcb_util_renderutil_jll"),
+    Dependency("xkbcommon_jll"),
+    Dependency("Libglvnd_jll"),
+    Dependency("Fontconfig_jll"),
+    Dependency("Glib_jll"),
+]
+
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"7")
+build_tarballs(ARGS, name, version, sources, script, platforms_macos, products_macos, dependencies)
```

Comment:
@staticfloat given #585, I think that having different products won't work in practice in our CI system

Reply:
I think we're going to try and merge them, fail, and things are going to go bonkers.  We probably need to tweak json output a bit by keying everything in the json file by platform.

Reply:
@barche Can you try wrapping these calls to `build_tarballs()` in conditionals with `should_build_platform()`?  Check out what we do in `Git`: https://github.com/JuliaPackaging/Yggdrasil/blob/master/G/Git/build_tarballs.jl#L81-L86

Reply:
OK, that should work I think. 🤞

BTW, it's OK to do the `expand_cxxstring_abis` only on the non-Mac platforms, right?

Reply:
As long as you're certain you're using `clang++` for all C++ code and not `g++`, yes.

Reply:
I guess more precisely; as long as you're certain you're not using `libstdc++` as your backing C++ library, and are instead using `libc++` (which should be default on MacOS).